src/api/dependency-discovery/src/dependency-list.js

```diff
@@ -1,14 +1,52 @@
 const { readFile } = require('fs/promises');
 const { join } = require('path');
 const { cwd } = require('process');
+const { getPackument } = require('query-registry');
+
+const dependenciesGlobal = {};
```

Comment:
I suggest a nit to avoid having a global variable here

```js
const getDependencies = (function () {
  let dependencies = null;
  return async () => {
    if (!dependencies) {
      dependencies = {};
      console.log('reading');
      const dependencyList = await readFile(join(cwd(), 'deps.txt'), 'utf8');

      // The order of the alternatives is important!
      // The regex engine will favor the first pattern on an alternation
      // even if the other alternatives are subpatterns
      dependencyList
        .split(/\r\n|\n|\r/)
        .filter((line) => line !== '')
        .forEach((dependencyName) => {
          dependencies[dependencyName] = null;
        });
    }
    return dependencies;
  };
})();
```
This pattern makes the value of `dependencies` persist, in the meantime hides it from the global.

Reply:
I was thinking of using a closure, but I don't know if this is a pattern that everybody is familiar with, and I didn't want to introduce something that would complicate other people's understanding of the code. Otherwise, I agree with this suggestion.

Reply:
It is a common pattern.